### PR TITLE
Add separable convolution

### LINKED ISSUE #6630: separable convolution
The separable convolution is needed for implementing networks like Xception.

### PR DESCRIPTION
**Pull request by will-am**
Resolve #6630

### CODE REVIEW

File: python/paddle/trainer_config_helpers/networks.py [hunk 1]
```diff
@@ -435,6 +435,86 @@ def __extend_list__(obj):
...
+    :type param_attr: ParameterAttribute
+    :param shared_bias: see img_conv_layer for details.
+    :type shared_bias: bool
+    :param layer_type: see img_conv_layer for details.
+    :type layer_type: bool
+    :return: layer's output
+    :rtype: LayerOutput
+    """
+    __depthwise_conv__ = img_conv_layer(
+        name="%s_depthwise_conv" % name,
```
Comment (NHZlX) [inline python/paddle/trainer_config_helpers/networks.py]: If the user doesn't set the name, there must be bug.

File: python/paddle/trainer_config_helpers/networks.py [hunk 2]
```diff
@@ -435,6 +435,86 @@ def __extend_list__(obj):
...
+        filter_size=filter_size,
+        stride=stride,
+        padding=padding,
+        act=LinearActivation(),
+        bias_attr=bias_attr,
+        param_attr=param_attr,
+        shared_biases=shared_bias,
+        layer_type=layer_type)
+    __pointwise_conv__ = img_conv_layer(
+        name="%s_pointwise_conv" % name,
```
Comment (NHZlX) [inline python/paddle/trainer_config_helpers/networks.py]: Ditto

File: python/paddle/trainer_config_helpers/networks.py (hunk 1 above)
Comment (NHZlX) [reply python/paddle/trainer_config_helpers/networks.py]: Depthwise separable convolution used in many models(https://github.com/PaddlePaddle/Mobile/blob/develop/models/standard_network/mobilenet.py). But It seems that it usually has a batch norm layer after every conv, how about set an option ?

Comment (will-am) [reply python/paddle/trainer_config_helpers/networks.py]: Thank you very much for your reviews.

1. the `@wrap_name_default("separable_conv")` at the very beginning ensures the name will always be set even the users do not provide.

2. I surveyed the separable conv in TF and MXNet, there is no batch-norm between depthwise conv and pointwise conv. The batch norm comes after each of the separable_conv. The demo code you provided also contains no batch-nom between depthwise and pointwise conv:

File: python/paddle/trainer_config_helpers/networks.py (hunk 2 above)
Comment (will-am) [reply python/paddle/trainer_config_helpers/networks.py]: Same as above.
